### ISSUE TITLE
Add CTA Embed Tag

### ISSUE THREAD

**Issue by benhalpern** [forem/forem#19378]
## Is this feature related to a problem?  Please describe.

Billboards are flexible space which allow use of all the same publishing tooling that authors have, such as different H tags, bold, italic, and also embeds.

Embeds are the way to allow a fully expressive area, and one key functionality we have been building towards is a simple new embed (aka liquid tag) called a “CTA”.

Embeds are an area in need of an overhaul in terms of how they are rendered in general, but this is a task where we should follow existing patterns. Options for re-writing how embeds are rendered is something we can explore as we invest more in this area.

### What billboards offer now:

<img width="359" alt="Screen Shot 2023-04-21 at 4 40 01 PM" src="https://user-images.githubusercontent.com/3102842/233730576-8a6a2c3a-3ee3-4701-a6e6-5102b1b878e4.png">

### What "Read full post" could look like with this feature:

<img width="255" alt="Screen Shot 2023-04-21 at 4 40 15 PM" src="https://user-images.githubusercontent.com/3102842/233730596-300f1231-1b3c-47ed-bf10-1c07a6ada7c8.png">

## Describe the solution you’d like

The CTA embed is designed to give creators of billboards the use of a button link, just like we have natively hardcoded elsewhere on the platform. As with other embeds, this will also give *any author* use of this same rich link in their posts if they choose to use it, which is a neat new feature.

The embed should reflect current site design, but may have a slightly different look in different context (i.e. font-size, padding, width, etc.)

The embed will require a dynamic liquid format similar to the “Detail Tag”.

It should look like this:

```
{% cta https://my-link.com/link %}
Click my awesome button!
{% endcta %}
```

This will render HTML in the above fashion.

While this is meant to mimic site style, we should likely not use utility classes, and instead mimic other liquid tags and create a class such as `.ltag_cta`. This will give us more flexibility to define slight variations in different contexts as needed.

The button should be full width in most contexts, but possibly have a have a design-defined max width so it is not full width in posts, etc.

Specs:

- Only plan text should be allowed in the text space, no tags should be rendered!
- There should be a limit of 128 characters in the text space (same as what we allow for article titles, to limit abuse of the design by users).

## Definition of Done

- [ ] Implement tag
- [ ] Add to editor embed guidelines and documentation
- [ ] Finalize design edge cases

## Additional Context

This should be accomplishable swiftly by following patterns established with tags such as `app/liquid_tags/details_tag.rb` 

As noted above, there is technical debt in embeds where we cannot easily update embed designs without having to re-save old posts. This is something we need to fix in the long run. We’ll need some technical discussions to make this happen, and this task can help us focus some of that discussion.

**Comment by Ridhwana** [comment]
> As noted above, there is technical debt in embeds where we cannot easily update embed designs without having to re-save old posts. This is something we need to fix in the long run.

Does this mean that as part of the task we want to resave old posts?

**Comment by mirie** [comment]
@anujbhavsar96 we would like to understand how the CTA should look in different Billboard placements, Posts, and Comments (for example should it be full-width or not and it is centered) for this point:  `Finalize design edge cases`

**Comment by benhalpern** [comment]
> When we define other variations in the future, do we envision that the classnames may be something like .ltag_cta--primary,.ltag_cta--secondary etc.

Yes, that is how we'd do this. We don't necessarily need this functionality now, but once this could be _extended_ I think the extension would be a second argument, such as this:

```
{% cta https://my-link.com/link secondary %}
Click my awesome button!
{% endcta %}
```

**Comment by anujbhavsar96** [comment]
This is how I think CTAs should look on billboards. 

What do you think, @benhalpern ?

<img width="1320" alt="Screenshot 2023-04-25 at 9 33 30 AM" src="https://user-images.githubusercontent.com/84626054/234172813-a8967b6e-c87e-4e85-be04-346cf8ba684f.png">

**Comment by benhalpern** [comment]
Okay @anujbhavsar96 this looks good to me.

To summarize:

- Sidebar are full width all the time.
- Other locations are just inline block with the described padding, but are full width at mobile width breakpoint.

We will want _some kind of horizontal padding_ for the full width variation, so that the text looks good, but I imagine that padding could be pretty minimal *i.e. `var(--su-2)` in that case.

I think this is all pretty straightforward, but will need to be double checked at PR time to make sure this is all reflected in the design.
